assets/js/base/hooks/checkout/use-checkout-notices.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import useCheckoutContext from '@woocommerce/base-context/checkout-context';
+
+const useCheckoutNotices = () => {
+	const { notices, updateNotices } = useCheckoutContext();
+	const addNotice = ( notice ) => {
+		updateNotices( ( originalNotices ) => [ ...originalNotices, notice ] );
+	};
+	const removeNotice = ( notice ) => {
+		// @todo...figure out how notices are saved - might need unique ids?
```

Comment:
Maybe a `clearAllNotices` will be enough for all use cases. The normal use case is that after displaying an error, you let the user retry ad then the error notice dissapears while the payment is processing again.